cmd/runtimecfg/node-ip.go

```diff
@@ -145,6 +145,20 @@ func getSuitableIPs(retry bool, vips []net.IP, preferIPv6 bool) (chosen []net.IP
 		if len(vips) > 0 {
 			chosen, err = utils.AddressesRouting(vips, utils.ValidNodeAddress)
 			if len(chosen) > 0 || err != nil {
+
+				// If using IPv6, verify that the choosen address isn't tentative
+				// i.e. we can actually bind to it
+				if net.IPv6len == len(chosen[0]) {
+					_, err := net.Listen("tcp", "["+chosen[0].String()+"]:10010")
```

Comment:
Binding to a hard-coded port seems a little hacky - although we know that the way things are deployed on startup this should run prior to `crio.service`, this would fail if someone runs the node-ip check manually after it's running, right?

I'm wondering if we should either use a random high port, or ideally check the connection flags e.g can we determine if the state is/isn't tentative via the golang rtnetlink bindings?

https://pkg.go.dev/github.com/vishvananda/go-netlink/rtnetlink/addr#Flags

Reply:
I can remove the port number and a random on is used, I didn't do this as it may pick a port another service wants, but its so short lived I guess it should be ok

I can Look into checking the flags